### DISCUSSION TITLE
How to uprade from 0.0.23 to 0.0.26

### DISCUSSION THREAD

**Discussion by pickeringar** [CJNE/ha-myenergi#499] [Q&A]
I am currently on 0.0.23 having encountered the entity duplication fault  #430 introduced in 0.0.24 and rolling back.

I now want to update to 0.0.26, but seeking advise on best way to do so and avoid any duplicate entities?

Any guidance? I am on RPi4

**Comment by HLFCode** [comment]
I had this problem going from 0.0.18-0.0.26
After installing the 0.0.26 version I got around it by:

- Open the myenergi integration and select the hub
- delete all the unavailable [i.e. original] sensors (through their settings)
- repeat for zappi (and presumably all other devices)
- delete the myenergi integration completely (this deletes the new ..._2 sensors)
- re-add the myenergi integration (needs hub s/n and password/key)

It seems it then creates the original sensor names (without ..._2) and the history is surprisingly still there (at least it was for me)

After this I didn't need to change any other configs/dashboards/automations to use the 0.0.26 version
Note I don't use HACS, I just copy the files to /config/custom_components
Hope it works for you @pickeringar

**Comment by pickeringar** [comment]
@HLFCode ....Thank you so much for your reply, confirming that the upgrade **does re-introduce the duplicate entries** that first arose in 0.0.24 and the steps  you took to “correct” matters after the upgrade.

As you ended up deleting the integration and re-adding it, I might try deleting the existing installation and then installing the latest release first to see if that also avoids the duplication. What I don’t want to do is loose my existing data.

**Comment by HLFCode** [comment]
Take a backup and be prepared to allocate the time to use it :-)
Deleting the original (with good data historic data behind it) might be OK as deleting entries singly didn't delete my historic data when they were re-created by the new install.
I have along disk write interval on recorder so maybe I was between writes when I deleted/recreated...
Good Luck
